### ISSUE TITLE
How can I model time series shorter than 512 timesteps?

### ISSUE THREAD

**Issue by GoerkeA** [moment-timeseries-foundation-model/moment-research#4]
Hi, i'd like to try your forecasting model on time series with less than T=512 timesteps but at this time i can't.

It works perfectly when i use 512 timesteps in my data, but each time i change with a lower or a higher value, the model returns an error about `RuntimeError: mat1 and mat2 shapes cannot be multiplied`

For example, with your example in *moment/tutorial/forecasting.ipynb*, when i change 
```python
x = torch.randn(16, 1, 512)
output = model(x)
```
to 
```python
x = torch.randn(16, 1, 256)
output = model(x)
```

I have the following error :

```python
File /anaconda/envs/foundation-models/lib/python3.10/site-packages/momentfm/models/moment.py:90, in ForecastingHead.forward(self, x, input_mask)
     88 def forward(self, x, input_mask: torch.Tensor = None):
     89     x = self.flatten(x)
---> 90     x = self.linear(x)
     91     x = self.dropout(x)
     92     return x

File /anaconda/envs/foundation-models/lib/python3.10/site-packages/torch/nn/modules/module.py:1501, in Module._call_impl(self, *args, **kwargs)
   1496 # If we don't have any hooks, we want to skip the rest of the logic in
   1497 # this function, and just call forward.
   1498 if not (self._backward_hooks or self._backward_pre_hooks or self._forward_hooks or self._forward_pre_hooks
   1499         or _global_backward_pre_hooks or _global_backward_hooks
   1500         or _global_forward_hooks or _global_forward_pre_hooks):
-> 1501     return forward_call(*args, **kwargs)
   1502 # Do not call functions when jit is used
   1503 full_backward_hooks, non_full_backward_hooks = [], []

File /anaconda/envs/foundation-models/lib/python3.10/site-packages/torch/nn/modules/linear.py:114, in Linear.forward(self, input)
    113 def forward(self, input: Tensor) -> Tensor:
--> 114     return F.linear(input, self.weight, self.bias)

RuntimeError: mat1 and mat2 shapes cannot be multiplied (16x32768 and 65536x7)
```

It seems to be a topic you think about because it's mentionned on your *Frequently Asked Questions* word, but it's empty.

![image](https://github.com/user-attachments/assets/701e1efd-feff-4321-b02e-9fd492574dd7)

**Do you know how to change this constraint ?**

**Comment by mononitogoswami** [comment]
Hi Goerke, 
Thanks for your interest in MOMENT!

**TL;DR**: The model expects 512 time steps as input. One straightforward way to model shorter time series is by padding shorter time series such that they are 512 time steps in length. 

**Longer answer**: 
One thing to be careful of is to pad shorter time series upto a multiple of 8. For example, if a time series has length 30, I would recommend padding it to 32. This is because MOMENT operates on patches of length 8, and if any patch is partially observed, it ignores the entire patch. This way, if you did not pad the time series to length 32, you would miss 6 time steps worth of information, which is significant for smaller time series. 

However, you do not need to pad it all the way up to 512 time steps. I would recommend using the `input_mask` (https://github.com/moment-timeseries-foundation-model/moment/blob/90e7bacaea878050cfa5806a3b0e894a7a463dbb/momentfm/models/moment.py#L281) to prevent the model from attending to the padded time steps.  
 
Let me know if you have any more questions!

Best,
Mononito
